irohad/consensus/yac/impl/yac_gate_impl.cpp

```diff
@@ -49,11 +49,10 @@ namespace iroha {
             consensus_result_cache_(std::move(consensus_result_cache)),
             log_(logger::log("YacGate")) {
         block_creator_->on_block().subscribe(
-            [this](const auto &block) { this->vote(block); });
+            [this](auto block) { this->vote(*block); });
```

Comment:
As I understand, type of `block` here is pointer, then you dereference it, but in function below again wrap into pointer. Maybe, there's sense in passing the pointer directly?